### PR TITLE
Migrate contacts import job to new API

### PR DESCRIPTION
**Pull request by ezaquarii**
Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

- [x] Tests written

### CODE REVIEW

File: src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                        operations.insertContact(vCard)
+                    } else {
+                        operations.updateContact(vCard, ownContactMap[vCard])
+                    }
+                } else {
+                    operations.insertContact(vCard) // Insert All the contacts without name
+                }
+            }
+        } catch (e: Exception) {
+            logger.e(TAG, "${e.message}")
```
Comment (AndyScherzinger) [inline src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt]: shouldn't we rather log the whole exception as in stack trace?

File: src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        try {
+            contentResolver.openInputStream(uri).use { inputStream ->
+                val vCardList = ArrayList<VCard>()
+                vCardList.addAll(Ezvcard.parse(inputStream).all())
+                if (vCardList.size > 0) {
+                    vCard = vCardList[0]
+                }
+            }
+        } catch (e: IOException) {
+            logger.d(TAG, "${e.message}")
```
Comment (AndyScherzinger) [inline src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt]: same here, no logging of the StackTrace

**Comment by AndyScherzinger** [comment]
@ezaquarii Just two minor comments about exception logging 👍

File: src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    companion object {
+        const val TAG = "ContactsImportWork"
+        const val ACCOUNT_TYPE = "account_type"
+        const val ACCOUNT_NAME = "account_name"
+        const val VCARD_FILE_PATH = "vcard_file_path"
+        const val SELECTED_CONTACTS_INDICES = "selected_contacts_indices"
+    }
+
+    override fun doWork(): Result {
```
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt:58]: This is legacy code copied and converted to Kotlin. No changes to logic are intended at this stage to limit the risk.

[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt:58]: This is legacy code copied and converted to Kotlin. No changes to logic are intended at this stage to limit the risk.

File: src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                    if (!ownContactMap.containsKey(vCard)) {
+                        operations.insertContact(vCard)
+                    } else {
+                        operations.updateContact(vCard, ownContactMap[vCard])
+                    }
+                } else {
+                    operations.insertContact(vCard) // Insert All the contacts without name
+                }
+            }
+        } catch (e: Exception) {
```
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt:102]: This is legacy code copied and converted to Kotlin. No changes to logic are intended at this stage to limit the risk.

File: src/main/java/com/owncloud/android/ui/fragment/contactsbackup/ContactListFragment.java [hunk 5]
```diff
@@ -409,10 +399,10 @@ private void getAccountForImport() {
                     String name = cursor.getString(cursor.getColumnIndex(ContactsContract.RawContacts.ACCOUNT_NAME));
                     String type = cursor.getString(cursor.getColumnIndex(ContactsContract.RawContacts.ACCOUNT_TYPE));
 
-                    ContactAccount account = new ContactAccount(name, name, type);
+                    ContactsAccount account = new ContactsAccount(name, name, type);
```
[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/fragment/contactsbackup/ContactListFragment.java:402]: This is legacy code copied and converted to Kotlin. No changes to logic are intended at this stage to limit the risk.

[the remark this answers is not in the document]
Comment (ezaquarii) [reply src/main/java/com/owncloud/android/ui/fragment/contactsbackup/ContactListFragment.java:402]: Rename only. No logic chage intended at this tage so I propose we `wontfix` it.

**Comment by tobiasKaminsky** [comment]
> @ezaquarii Just two minor comments about exception logging +1

I would also vote for logging the whole exception.

File: src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt (hunk 2 above)
Comment (ezaquarii) [reply src/main/java/com/nextcloud/client/jobs/ContactsImportWork.kt:124]: https://github.com/nextcloud/android/pull/5835

**Comment by ezaquarii** [comment]
> > @ezaquarii Just two minor comments about exception logging +1
> 
> I would also vote for logging the whole exception.

Fixed here: https://github.com/nextcloud/android/pull/5835

**Pull request merged**